packages/app/src/app/pages/Sandbox/Editor/Workspace/items/Deployment/Netlify.tsx

```diff
@@ -0,0 +1,200 @@
+import { Button } from '@codesandbox/common/lib/components/Button';
+import { resolveDirectory } from '@codesandbox/common/lib/sandbox/modules';
+import getTemplate from '@codesandbox/common/lib/templates';
+import { observer } from 'mobx-react-lite';
+import React, { useEffect, useState } from 'react';
+import LinkIcon from 'react-icons/lib/fa/external-link';
+import Cogs from 'react-icons/lib/fa/cogs';
+import LightningIcon from 'react-icons/lib/md/flash-on';
+
+import DeploymentIntegration from 'app/components/DeploymentIntegration';
+import NetlifyLogo from 'app/components/NetlifyLogo';
+import { useSignals, useStore } from 'app/store';
+import getNetlifyConfig from 'app/utils/getNetlifyConfig';
+
+import { WorkspaceInputContainer, WorkspaceSubtitle } from '../../elements';
+
+import {
+  Deploys,
+  Deploy,
+  Name,
+  Link,
+  DeploysWrapper,
+  Wrapper,
+  ButtonContainer,
+} from './elements';
+
+const getFunctionDir = sandbox => {
+  try {
+    return resolveDirectory(
+      getNetlifyConfig(sandbox).functions,
+      sandbox.modules,
+      sandbox.directories
+    );
+  } catch (e) {
+    return [];
+  }
+};
+
+export const Netlify = observer(() => {
+  const {
+    deployment: { getNetlifyDeploys, deployWithNetlify },
+    modalOpened,
+  } = useSignals();
+  const { deployment, editor } = useStore();
+
+  const [isVisible, setVisibility] = useState(false);
+
+  useEffect(() => getNetlifyDeploys(), []); // eslint-disable-line
+
+  const template = getTemplate(editor.currentSandbox.template);
+  const functionDirectory = getFunctionDir(editor.currentSandbox);
+
+  const functions = editor.currentSandbox.modules.filter(
+    m => m.directoryShortid === functionDirectory.shortid
+  );
+
+  return (
+    template.netlify !== false && (
+      <Wrapper loading={deployment.deploying}>
+        <WorkspaceInputContainer style={{ marginTop: '1rem', marginBottom: 0 }}>
+          <DeploymentIntegration
+            beta
+            color="#FFFFFF"
+            deploy={() => deployWithNetlify()}
```

Comment:
@Saeris This can be simplified

```suggestion
            deploy={deployWithNetlify}
```

Reply:
Make sure to test whether it works, sometimes there's some iffy `this` or given params.

Reply:
Yeah a lot of the time where I'm doing this, there used to be a class method or something similar where there was a named function/method that wrapped a signal call. This is probably fine but it should be tested of course.